code/game/machinery/portable_turret.dm

```diff
@@ -102,16 +102,25 @@
 	//Sets up a spark system
 	spark_system = bind_spark(src, 5)
 
+	var/area/control_area = get_area(src)
+	if(istype(control_area))
+		var/obj/machinery/turretid/aTurretID = locate() in control_area

```

Comment:
`locate()` can return null, for one. Gotta check for that.

Second, there can be multiple turret controls per area. You want to call `turretModes` for all of them. They're also held in a neat little list, so you can do this with minimal expense:
```dm
for (var/TID in control_area.turret_controls)
	var/obj/machinery/turretid/turret_id = TID
	turret_id.turretModes()
```

Reply:
Ideally we wouldn't want to have multiple turret controls in same area. I haven't checked what happens when multiple of them are present, but if you want me to make them all be okay together I will change it into for loop.

Reply:
I'm pretty sure there are places with multiple turret controls per area. Or were. Anyways, it's a feature and a list is used, so for-loop it.